core/kernel/dt.c

```diff
@@ -125,6 +125,39 @@ int dt_map_dev(const void *fdt, int offs, vaddr_t *base, size_t *size)
 	return 0;
 }
 
+int dt_map_secure_dev(const void *fdt, int offs, vaddr_t *base, size_t *size)
```

Comment:
This duplicates most of the code in `dt_map_dev()`. Would it be possible to refactor the two functions to use a common helper function or something?

Reply:
Sounds good to me.

Reply:
I can try the proposed method.